@navikt/core/react/src/overlays/action-menu/ActionMenu.stories.tsx

```diff
@@ -0,0 +1,852 @@
+                  <ActionMenu.Item
+                    onSelect={() => console.log("Nested Subitem 2 clicked")}
+                  >
+                    Nested Subitem 2
+                  </ActionMenu.Item>
+                </ActionMenu.SubContent>
+              </ActionMenu.Sub>
+            </ActionMenu.SubContent>
+          </ActionMenu.Sub>
+
+          <ActionMenu.Item onSelect={() => console.log("Item 3 clicked")}>
+            Item 3
+          </ActionMenu.Item>
+        </ActionMenu.Content>
+      </ActionMenu>
+    );
+  },
+  decorators: [DemoDecorator],
+};
+
+export const Disabled: Story = {
+  render: (props) => {
+    return (
+      <ActionMenu open={props.open}>
+        <ActionMenu.Trigger>
+          <button>Open action</button>
+        </ActionMenu.Trigger>
+        <ActionMenu.Content>
+          <ActionMenu.Item
+            onSelect={() => console.log("Item 1 clicked")}
+            disabled
+            shortcut="T+W"
+          >
+            Item 1
+          </ActionMenu.Item>
+          <ActionMenu.Sub>
+            <ActionMenu.SubTrigger disabled>Submenu 1</ActionMenu.SubTrigger>
+            <ActionMenu.SubContent>
+              <ActionMenu.Item
+                onSelect={() => console.log("Subitem 1 clicked")}
+              >
+                Subitem 1
+              </ActionMenu.Item>
+            </ActionMenu.SubContent>
+          </ActionMenu.Sub>
+
+          <ActionMenu.CheckboxItem checked disabled shortcut="T+W">
+            Checkbox disabled
+          </ActionMenu.CheckboxItem>
+          <ActionMenu.CheckboxItem>Checkbox 2</ActionMenu.CheckboxItem>
+          <ActionMenu.RadioGroup value="1" label="Radiogroup">
+            <ActionMenu.RadioItem disabled value="1">
+              Radio disabled
+            </ActionMenu.RadioItem>
+            <ActionMenu.RadioItem value="2">Radio</ActionMenu.RadioItem>
+          </ActionMenu.RadioGroup>
+        </ActionMenu.Content>
+      </ActionMenu>
+    );
+  },
+  decorators: [DemoDecorator],
+};
+
+export const TriggerWithTooltip: Story = {
+  render: () => {
+    return (
+      <ActionMenu>
+        <Tooltip content="Tooltip!">
+          <ActionMenu.Trigger>
+            <button>Open action</button>
+          </ActionMenu.Trigger>
+        </Tooltip>
+        <ActionMenu.Content>
+          <ActionMenu.Item onSelect={() => console.log("Item 1 clicked")}>
+            Item 1
+          </ActionMenu.Item>
+          <ActionMenu.Item onSelect={() => console.log("Item 2 clicked")}>
+            Item 2
+          </ActionMenu.Item>
+          <ActionMenu.Item onSelect={() => console.log("Item 3 clicked")}>
+            Item 3
+          </ActionMenu.Item>
+        </ActionMenu.Content>
+      </ActionMenu>
+    );
+  },
+  decorators: [DemoDecorator],
+};
+
+/**
+ * TODO: Bugs
+ * - When keydown "space" on open modal button, the modal is closed instantly.
+ * Unsure if this is because of the keydown-event repeats or if its caused by eventbubbling
+ */
```

Comment:
First with "Enter" here, then "Space"

Enter works just like with a regular button. Using space triggers the event onKeyDown, then closes modal onKeyUp 😵😵😵 Unsure what the best way would be to handle this without lots of smartness, could just trigger onSelect onKeyUp to avoid both of the Enter and Space issue

https://github.com/user-attachments/assets/5c6b306d-1ce9-4a74-950c-d26b0f050d0f

Reply:
Ah, it only happens in Chrome since autofocus doesn't work there 😅

onKeyUp may be a solution. What was the reason we couldn't use the button element again?

Reply:
Ended up with this nifty solution
https://github.com/navikt/aksel/pull/3081/commits/8237929277b9bbc456cfb7dc7ec692cec31bea72

[MDN](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/repeat)

event.repeat just checks if the key is held down and is activating multiple times. If that is the case we can assume that they don't want to auto-close modal. I did the same in MenuItem keydown to avoid auto-selecting first item when opening the menu for the first time. Hope this solves it, but might be some edgecases i have forgotten here 🤔

Reply:
I'm afraid it still happens with Space, and in Safari also with Enter, but only if you hold it down. I think with Space it happens onKeyUp.

Also, the fix will only work if you use the built-in close button in Modal.Header, not if you set `closeButton: false` and add your own close button.

I guess this is the punishment of trying to re-implement native behavior 😅

I think the root of the problem is that onSelect triggers onKeyDown. Is it possible to change to onKeyUp or onClick? (Even if we still have to keep some logic in onKeyDown).

Reply:
Another way to fix it could be to force the focus to the Modal itself when it opens (like autoFocus is supposed to do, but only works in FF for now), but that feels hacky. It would be better to find a fix on the "ActionMenu-side", since the problem might affect other components (or maybe even native stuff) as well that we haven't discovered yet.